stan/math/opencl/multiply.hpp

```diff
@@ -37,6 +37,28 @@ inline auto multiply(const matrix_cl& A, const matrix_cl& B) {
     temp.zeros();
     return temp;
   }
+  if (A.rows() == 1 && triangular_view_A == TriangularViewCL::Entire
```

Comment:
Why do we need to check for triangularity here? The general matrix multiply is not faster here even with triangular optimizations, right? Is this only to ensure correct results for cases when someone wants to do a triangular multiply with non-zeroed out matrices and a row_vector/vector?

Reply:
Yes. If needed the new kernels could be extended to correctly do triangular multiplication.

Reply:
That would be great. I am not sure if these triangular cases come up often, but it would be nice to have these speedups for them.